Objects/longobject.c

```diff
@@ -4343,8 +4371,8 @@ long_invert(PyLongObject *v)
 {
     /* Implement ~x as -(x+1) */
     PyLongObject *x;
-    if (Py_ABS(Py_SIZE(v)) <=1)
-        return PyLong_FromLong(-(MEDIUM_VALUE(v)+1));
+    if (IS_MEDIUM_VALUE(v))
+        return PyLong_FromLong(-(medium_value(v)+1));
```

Comment:
Would it make sense to update this to be consistent with the other bitwise operators and use `_PyLong_FromLong(~medium_value(v))` here?

I think the `-(x + 1)` pattern came from days when we worried too much about alternative integer representations allowed by the C spec (sign-magnitude, ones' complement, two's complement with a trap representation, etc.). But since then other parts of the codebase have been happy to assume that all integer types are two's complement, no padding bits, no trap representation, so I think it's safe to do so here. (And your changes to the other bitwise operations already assume two's complement and no trap representation.)

Reply:
(Sorry, I meant `_PyLong_FromSDigit`, not `_PyLong_FromLong`; edited the message above to fix.)